Docs(config): Add output.strictModuleErrorHandling (#4710)
webpack/webpack.js.org: src/content/configuration/output.mdx

T> Typically you don't need to change `output.sourcePrefix`.

> ## `output.strictModuleExceptionHandling`

Reviewer: It's deprecated but still available to use. So let's add a warning and keep it at the moment until it's removed from webpack.

`boolean`

When using `libraryTarget: "umd"`, setting `output.umdNamedDefine` to `true` will name the AMD module of the UMD build. Otherwise an anonymous `define` is used.

```javascript
module.exports = {
  //...
  output: {
    umdNamedDefine: true,
  },
};
```

## `output.strictModuleErrorHandling`

`boolean`

> Handles error in module loading correctly at a performance cost. This will handle module error compatible with the EcmaScript Modules spec.

Reviewer: ```suggestion
Handle error in module loading as per EcmaScript Modules spec at a performance cost.
```
